src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,106 @@
+        taskList.add(task);
+        final int listLength = taskList.size();
+        //To check that task was added successfully
+        final Task mostRecentTask = taskList.get(listLength - 1);
+        Ui.printAddTaskMessage(listLength, mostRecentTask);
+    }
+
+    public void addTaskWithoutMessage(Task task) throws DukeException {
+        if (task.getDescription().equals("")) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_NO_DESCRIPTION);
+        }
+        taskList.add(task);
+    }
+
+
+    public void deleteTask(String input) throws DukeException, NumberFormatException {
+        if (input.equals("")) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_NO_TASK_NUMBER);
+        }
+        int taskNumber = Integer.parseInt(input.trim());
+        if (isMissingTask(taskNumber)) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_INVALID_TASK_NUMBER);
+        }
+        deleteExistingTask(taskNumber);
+    }
+
+    private void deleteExistingTask(int taskNumber) {
+        final int taskIndex = taskNumber - 1;
+        Ui.printDeleteTaskSuccessMessage(taskList.get(taskIndex));
+        taskList.remove(taskIndex);
+    }
+
+
+    public void markTaskAsDone(String input) throws DukeException, NumberFormatException {
+        if (input.equals("")) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_NO_TASK_NUMBER);
+        }
+        int taskNumber = Integer.parseInt(input.trim());
+        if (isMissingTask(taskNumber)) {
+            throw new DukeException(ExceptionMessages.EXCEPTION_INVALID_TASK_NUMBER);
+        }
+        markExistingTaskAsDone(taskList.get(taskNumber - 1));
+    }
+
+    private boolean isMissingTask(int taskNumber) {
+        final int listLength = taskList.size();
+        return taskNumber <= 0 || taskNumber > listLength;
+    }
+
+    private void markExistingTaskAsDone(Task task) {
+        task.markTaskAsDone();
+        Ui.printMarkTaskSuccessMessage(task);
+    }
+
+    public void listAllTasks() {
+        Ui.printAllTasks(taskList);
+    }
+
+    public String toFile() {
+        StringBuilder fileContent = new StringBuilder();
+        for (int i = 0; i < taskList.size(); i++) {
+            Task taskToConvert = taskList.get(i);
+            if (taskToConvert instanceof Todo) {
+                fileContent.append("T|").append(taskToConvert.isDone ? '1' : '0').append('|').append(taskToConvert.description);
+            } else if (taskToConvert instanceof Deadline) {
+                fileContent.append("D|").append(taskToConvert.isDone ? '1' : '0').append('|').append(taskToConvert.description)
+                        .append('|').append(((Deadline) taskToConvert).getBy());
+            } else if (taskToConvert instanceof Event) {
+                fileContent.append("E|").append(taskToConvert.isDone ? '1' : '0').append('|').append(taskToConvert.description)
+                        .append('|').append(((Event) taskToConvert).getAt());
```

Comment:
Some the lines here are too long (>120 chars per line). You could consider formatting the strings differently to shorten the line, see[ Java string format() ](https://www.javatpoint.com/java-string-format)method for examples.